src/microvis/backend/pygfx/_view.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import pygfx
+
+from microvis import core
+
+from ._node import Node
+
+if TYPE_CHECKING:
+    from microvis import _types
+
+    from ._camera import Camera as CameraAdaptor
+
+
+class View(Node, core.view.ViewBackend):
+    """View interface for pygfx Backend."""
+
+    # _native: scene.ViewBox
+    # TODO: i think pygfx doesn't see a view as part of the scene like vispy does
+    _camera: CameraAdaptor
+
+    def __init__(self, view: core.View, **backend_kwargs: Any) -> None:
+        # FIXME:  hardcoded camera and scene
+        self.scene = pygfx.Scene()
+
+    # XXX: both of these methods deserve scrutiny, and fixing :)
+    def _viz_set_camera(self, cam: core.Camera) -> None:
+        if not isinstance(cam.native, pygfx.Camera):
+            raise TypeError(f"cam must be a pygfx.Camera, got {type(cam.native)}")
+        self._camera = cam.backend_adaptor()
+
+    def _viz_set_scene(self, scene: core.Scene) -> None:
```

Comment:
i'd say this was the main observation in trying to implement this.  This is the pattern I used over in the vispy adaptor, but it's troublesome.  The general "issue" here is that we want to tell an adaptor to set an object, and we pass it a `core` object, but it needs to make sure there is a proper `backend` object instantiated.  So that `if not isinstance(scene.native,` line there is really quite overloaded.  it does this:

1. core calls `_viz_set_scene` passing a `core.Scene` object
2. the adaptor accesses `scene.native` (i.e. asking for an object from the backend library)
3. `scene.native` is a property that actually calls `scene.backend_adaptor()._vis_get_native()` ... thereby creating an adaptor if one doesn't yet exist.

`3` is definitely an important step.  and I'm not sure it's "wrong" that it happens right at this moment (it's part of how the model can be "rehydrated", kinda recursively, only once it's necessary)... but `2` really seems to be the bit that could be made more explicit.  Specifically: how does an adaptor object say "please give me the object from my own backend library".  Maybe a better pattern here would be for 2 to look something more like:

```py
# the adaptor can ask for a backend adaptor (but *not* `.native`)
# furthermore, it should probably be explicit about specifying it's own name
adaptor = scene.backend_adaptor('my_backend_name')
# once it has the adaptor, it can access anything it needs on it, since
# at that point it's an internal "adaptor" implementation
adaptor._my_object
```

@alisterburt  ... that feels like what you were suggesting by saying "adaptors don't use `.native`"  yes?  thoughts?

Reply:
yep 100% with you, feels much nicer that way - good to untangle these knots early!